vlib/v/checker/tests/import_mod_sub_duplicate_as_mod.out

```diff
@@ -0,0 +1,5 @@
+vlib/v/checker/tests/import_mod_sub_duplicate_as_mod.vv:1:1: error: project must include a `main` module or be a shared library (compile with `v -shared`)
+    1 | module json2
+      | ^
+    2 | 
+    3 | import x.json2 as json
```

Comment:
This non-erroring test tests a import of a sub-module, so it was moved from `vlib/v/checker/tests/import_mod_duplicate_as_mod.out` to `vlib/v/checker/tests/import_mod_sub_duplicate_as_mod.out`

A non-erroring test for an alias import that is not a sub module, would reveal another bug, and in an ouput we don't want.
```v
module json

import json as vjson

numbers := {
	'one':   1
	'two':   2
	'three': 3
}
out := vjson.encode(numbers)
println(out)
```
Imports used like this will result in unkown function errors of the module that was imported as alias.
```
import.v:10:14: error: unknown function: json.encode 
    8 |     'three': 3
    9 | }
   10 | out := vjson.encode(numbers)
```

It's a fairly rare case needing to do an import as alias to use it in a module with the same name and probably neglectable. But it should be tracked in a separate issue.

Reply:
Forbidding any kind of import that has the same name - weather it's aliased or not - would definitely reduce headache covering all corners of this topic.

But maybe this solution would be to simple.
Preserving the possibility to alias an import when it has the same name, was in the mind that it's not a self import but when it's the import of another module or thrid party module with the same name. 

Is this restriction likely the wanted solution or should we ask the community about it?